src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IForLoopStatement.cs

```diff
@@ -35,10 +35,9 @@ static void Main()
       Left: ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Int32) (Syntax: 'i')
       Right: ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 3) (Syntax: '3')
   Before:
-      IVariableDeclarationStatement (1 declarations) (OperationKind.VariableDeclarationStatement) (Syntax: 'i = 0')
```

Comment:
>IVariableDeclarationStatement (1 declarations) (OperationKind.VariableDeclarationStatement) (Syntax: 'i = 0') [](start = 6, length = 109)

It is not clear why did we lose the statement, feels wrong.

Reply:
before any BoundLocalDeclaration was converted to IVariableDeclarationStatement, now only BoundLocalDeclaration that has LocalDeclarationStatementSyntax as Syntax converted to Statement, and others (which has VariableDeclaratorSyntax as Syntax) will be converted to IVariableDeclaration.

Reply:
> before any BoundLocalDeclaration was converted to IVariableDeclarationStatement, now only BoundLocalDeclaration that has LocalDeclarationStatementSyntax as Syntax converted to Statement, and others (which has VariableDeclaratorSyntax as Syntax) will be converted to IVariableDeclaration.

Let's open a follow up issue and bring this to the design meeting. Initializers are also not statements (see the next unit-test), but we add expression statement for them. Why not do this for declarations?

Reply:
done - https://github.com/dotnet/roslyn/issues/22208